tests/FakeItEasy.Analyzer.Tests/NonVirtualSetupAnalyzerTests.cs

```diff
@@ -0,0 +1,200 @@
+{
+    class TheClass
+    {
+        void TheTest()
+        {
+            var foo = A.Fake<Foo>();
+            A.CallTo(() => foo.Bar());
+        }
+    }
+
+    internal class Foo
+    {
+        internal void Bar()
+        {
+            throw new NotImplementedException();
+        }
+    }
+}
+
+";
+
+           VerifyCSharpDiagnostic(
+    test,
+    new DiagnosticResult
+    {
+        Id = DiagnosticDefinitions.NonVirtualSetup.Id,
+        Message =
+            "Non virtual member 'Bar' cannot be intercepted.",
+        Severity = DiagnosticSeverity.Warning,
+        Locations = new[] { new DiagnosticResultLocation("Test0.cs", 11, 28) }
+    });
+
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Be_Triggered_For_Non_Virtual_Member_WithArguments()
+        {
+            const string test = @"using System;
+
+//Temporary class to get non-virtual test correct
+    namespace FakeItEasy.Analyzer.Tests
+    {
+    class TheClass
+    {
+        void TheTest()
+        {
+            var foo = A.Fake<Foo>();
+            A.CallTo(() => foo.Bar(A<string>.Ignored));
+        }
+    }
+
+    internal class Foo
+    {
+        internal void Bar(string name)
+        {
+            throw new NotImplementedException();
+        }
+    }
+}";
+
+            VerifyCSharpDiagnostic(
+        test,
+        new DiagnosticResult
+        {
+            Id = DiagnosticDefinitions.NonVirtualSetup.Id,
+            Message =
+             "Non virtual member 'Bar' cannot be intercepted.",
+            Severity = DiagnosticSeverity.Warning,
+            Locations = new[] { new DiagnosticResultLocation("Test0.cs", 11, 28) }
+        });
+
+
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Correctly_Reflect_Member_Name()
+        {
+            //Test to ensure member name hasn't mistakenly been hardcoded
+
+            const string test = @"using System;
+
+//Temporary class to get non-virtual test correct
+    namespace FakeItEasy.Analyzer.Tests
+    {
+    class TheClass
+    {
+        void TheTest()
+        {
+            var foo = A.Fake<Foo>();
+            A.CallTo(() => foo.DifferentNameThanOtherTests(A<string>.Ignored));
+        }
+    }
+
+    internal class Foo
+    {
+        internal void DifferentNameThanOtherTests(string name)
+        {
+            throw new NotImplementedException();
+        }
+    }
+}";
+
+            VerifyCSharpDiagnostic(
+        test,
+        new DiagnosticResult
+        {
+            Id = DiagnosticDefinitions.NonVirtualSetup.Id,
+            Message =
+             "Non virtual member 'DifferentNameThanOtherTests' cannot be intercepted.",
+            Severity = DiagnosticSeverity.Warning,
+            Locations = new[] { new DiagnosticResultLocation("Test0.cs", 11, 28) }
+        });
+
+
+        }
+
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new NonVirtualSetupAnalyzer();
+        }
+
+    }
+
+}

```

Comment:
I'm enjoying the tests, @medmondson. Do you think there's value in adding tests (and possibly specific message, etc.) for static members? This can certainly be considered a "bonus", from my point of view.